UDPClientGUI/MyUDPClient.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Sockets;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace UDPCommGUI
+{
+    class MyUDPClient : IClientInterface
```

Comment:
Also implement IDisposable so you can dispose of it later

Reply:
Implemented for both server and client